src/cpp/dds/domain/DomainParticipantImpl.hpp

```diff
@@ -0,0 +1,352 @@
+    }
+
+    fastrtps::rtps::RTPSParticipant* rtps_participant()
+    {
+        return rtps_participant_;
+    }
+
+    const TypeSupport find_type(
+            const std::string& type_name) const;
+
+    const fastrtps::rtps::InstanceHandle_t& get_instance_handle() const;
+
+    // From here legacy RTPS methods.
+
+    const fastrtps::rtps::GUID_t& guid() const;
+
+    /**
+     * Get the participant attributes
+     * @return Participant attributes
+     */
+    inline const fastrtps::ParticipantAttributes& get_attributes() const
+    {
+        return att_;
+    }
+
+    std::vector<std::string> get_participant_names() const;
+
+    /**
+     * This method can be used when using a StaticEndpointDiscovery mechanism differnet that the one
+     * included in FastRTPS, for example when communicating with other implementations.
+     * It indicates the Participant that an Endpoint from the XML has been discovered and
+     * should be activated.
+     * @param partguid Participant GUID_t.
+     * @param userId User defined ID as shown in the XML file.
+     * @param kind EndpointKind (WRITER or READER)
+     * @return True if correctly found and activated.
+     */
+    bool new_remote_endpoint_discovered(
+        const fastrtps::rtps::GUID_t& partguid,
+        uint16_t userId,
+        fastrtps::rtps::EndpointKind_t kind);
+
+    bool get_remote_writer_info(
+        const fastrtps::rtps::GUID_t& writerGuid,
+        fastrtps::rtps::WriterProxyData& returnedInfo);
+
+    bool get_remote_reader_info(
+        const fastrtps::rtps::GUID_t& readerGuid,
+        fastrtps::rtps::ReaderProxyData& returnedInfo);
+
+    fastrtps::rtps::ResourceEvent& get_resource_event() const;
+
+    Subscriber* lookup_subscriber(
+            const fastrtps::rtps::InstanceHandle_t& handle);
+
+    Publisher* lookup_publisher(
+            const fastrtps::rtps::InstanceHandle_t& handle);
+
+    //! Remove all listeners in the hierarchy to allow a quiet destruction
+    void disable();
+
+private:
+
+    //!Participant Attributes
+    fastrtps::ParticipantAttributes att_;
+
+    //!RTPSParticipant
+    fastrtps::rtps::RTPSParticipant* rtps_participant_;
+
+    //!Participant*
+    DomainParticipant* participant_;
+
+    //!Participant Listener
+    DomainParticipantListener* listener_;
+
+    //!Publisher maps
+    std::map<Publisher*, PublisherImpl*> publishers_;
+    std::map<fastrtps::rtps::InstanceHandle_t, Publisher*> publishers_by_handle_;
+    mutable std::mutex mtx_pubs_;
+
+    PublisherQos default_pub_qos_;
+
+    //!Subscriber maps
+    std::map<Subscriber*, SubscriberImpl*> subscribers_;
+    std::map<fastrtps::rtps::InstanceHandle_t, Subscriber*> subscribers_by_handle_;
+    mutable std::mutex mtx_subs_;
+
+    SubscriberQos default_sub_qos_;
+
+    //!TopicDataType map
+    std::map<std::string, TypeSupport> types_;
+    mutable std::mutex mtx_types_;
+
+    class MyRTPSParticipantListener : public fastrtps::rtps::RTPSParticipantListener
```

Comment:
One level of indentation inside class

Reply:
To be discussed, because it is an inner class.